### PR TITLE
table: RenderTSV to render in tab-separated-values format

### LINKED ISSUE #276: Table - Allow `RenderCsv` to be formatted with tabs instead of comma
`RenderCsv` has the comma separator hardcoded.

I would like to render the CSV using tabs because it is easier to import the generated file in Google Sheets of Excel by just pasting the CSV content.

### PR DESCRIPTION
**Pull request by rafiramadhana**
## Proposed Changes
  - Add `RenderTSV() string` to `Writer` interface
  - Add `RenderTSV()` implementation

Fixes #276 .

@jedib0t @dinhani

### CODE REVIEW

**Comment by jedib0t** [comment]
Hey @rafiramadhana thanks a lot for contributing this feature. Code looks good, but I have one concern. I believe that columns in TSV format are not allowed to have `\t` and `\n` characters as it will break formatting. Can you figure out the best way to deal with them?

One option I'm thinking of is replacing `\t` with `\\t` and `\n` with `\\n` so that the reader can interpret it correctly, but I don't know if that is what is usually done.

**Comment by jedib0t** [comment]
@rafiramadhana It looks like you can double-quote columns which contain `\t` and `\n`. For example:

CSV:
```csv
"a","b	c","d
e"
````

TSV:
```tsv
a	"b	c"	"d
e"
```

Can you please make this change in the logic?

```golang
	if strings.ContainsAny(col, "\t\n") {
		out.WriteString(fmt.Sprintf("\"%s\"", col))
	} else {
		out.WriteString(col)
	}
```

**Comment by rafiramadhana** [comment]
> I believe that columns in TSV format are not allowed to have \t and \n characters as it will break formatting.

Yes, that is correct as mentioned in [Wikipedia](https://en.wikipedia.org/wiki/Tab-separated_values#:~:text=The%20IANA%20media%20type%20standard%20for%20TSV%20achieves%20simplicity%20by%20simply%20disallowing%20tabs%20within%20fields.%5B4%5D)

Sry, I haven't implement it because I rarely use a TSV

> @rafiramadhana It looks like you can double-quote columns which contain \t and \n.

Agree, I think we can start with this implementation of adding double-quote

**Comment by rafiramadhana** [comment]
> @rafiramadhana It looks like you can double-quote columns which contain \t and \n.

Do we need to escape the double-quote if the columns have double-quote?

```
asd"fgh    qwe    zxc

"asd""fgh"   qwe   zxc
```

This is similar with what is done in CSV ([rfc4180](https://datatracker.ietf.org/doc/html/rfc4180#:~:text=If%20double%2Dquotes%20are%20used%20to%20enclose%20fields%2C%20then%20a%20double%2Dquote%0A%20%20%20%20%20%20%20appearing%20inside%20a%20field%20must%20be%20escaped%20by%20preceding%20it%20with%0A%20%20%20%20%20%20%20another%20double%20quote.))

**Comment by jedib0t** [comment]
> > @rafiramadhana It looks like you can double-quote columns which contain \t and \n.
> 
> Do we need to escape the double-quote if the columns have double-quote?
> 
> ```
> asd"fgh    qwe    zxc
> 
> "asd""fgh"   qwe   zxc
> ```
> 
> This is similar with what is done in CSV ([rfc4180](https://datatracker.ietf.org/doc/html/rfc4180#:~:text=If%20double%2Dquotes%20are%20used%20to%20enclose%20fields%2C%20then%20a%20double%2Dquote%0A%20%20%20%20%20%20%20appearing%20inside%20a%20field%20must%20be%20escaped%20by%20preceding%20it%20with%0A%20%20%20%20%20%20%20another%20double%20quote.))

I'm not sure what the spec calls for. But pasting `"asd""fgh"	qwe	zxc` into Excel made it come out right. So I think you are on the right track. Let's go with this approach for now.

File: table/render_tsv.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+				out.WriteString(fmt.Sprint(hint.rowNumber))
+			}
+			out.WriteRune('\t')
+		}
+
+		if idx > 0 {
+			out.WriteRune('\t')
+		}
+
+		if strings.ContainsAny(col, "\t\n\"") || strings.Contains(col, "    ") {
```
Comment (rafiramadhana) [inline table/render_tsv.go:52]: @jedib0t  It looks like `\t` gets replaced by `"    "` (4 spaces) in [render_init.go#46](https://github.com/jedib0t/go-pretty/blob/main/table/render_init.go#L46)

So, that's why I'm adding `strings.Contains(col, " ")` here

Comment (jedib0t) [reply table/render_tsv.go:52]: Ok I'll have to refactor the original function. This may not be ideal behavior for the end-user.

**Comment by jedib0t** [comment]
Thanks @rafiramadhana for your contribution. Will cut a tag with this fix in a day or two.

**Comment by jedib0t** [comment]
Tag with this feature: https://github.com/jedib0t/go-pretty/releases/tag/v6.4.8

**Pull request merged**